src/gui/src/dbDescriptors.cpp

```diff
@@ -230,6 +230,206 @@ static odb::dbTechLayer* getLayerSelection(odb::dbTech* tech,
+  }
+  props.push_back({"Same-Net Rules", samenet_rules});
+
+  return props;
+}
+
+Selected DbTechDescriptor::makeSelected(std::any object) const
+{
+  if (auto tech = std::any_cast<odb::dbTech*>(&object)) {
+    return Selected(*tech, this);
+  }
+  return Selected();
+}
+
+bool DbTechDescriptor::lessThan(std::any l, std::any r) const
+{
+  auto l_tech = std::any_cast<odb::dbTech*>(l);
+  auto r_tech = std::any_cast<odb::dbTech*>(r);
+  return l_tech->getId() < r_tech->getId();
+}
+
+bool DbTechDescriptor::getAllObjects(SelectionSet& objects) const
+{
+  auto tech = db_->getTech();
+  if (tech == nullptr) {
+    return false;
+  }
+  objects.insert(makeSelected(tech));
+  return true;
+}
+
+//////////////////////////////////////////////////
+
+DbBlockDescriptor::DbBlockDescriptor(odb::dbDatabase* db) : db_(db)
+{
+}
+
+std::string DbBlockDescriptor::getName(std::any object) const
+{
+  auto block = std::any_cast<odb::dbBlock*>(object);
+  return block->getName();
+}
+
+std::string DbBlockDescriptor::getTypeName() const
+{
+  return "Block";
+}
+
+bool DbBlockDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  auto block = std::any_cast<odb::dbBlock*>(object);
+  bbox = block->getBBox()->getBox();
+  return true;
+}
+
+void DbBlockDescriptor::highlight(std::any object, Painter& painter) const
+{
+  auto block = std::any_cast<odb::dbBlock*>(object);
+
+  odb::dbBox* bbox = block->getBBox();
+  odb::Rect rect = bbox->getBox();
+  painter.drawRect(rect);
+}
+
+Descriptor::Properties DbBlockDescriptor::getProperties(std::any object) const
+{
+  auto block = std::any_cast<odb::dbBlock*>(object);
+
+  auto gui = Gui::get();
+
+  Properties props;
+  SelectionSet bterms;
+  for (auto bterm : block->getBTerms()) {
+    bterms.insert(gui->makeSelected(bterm));
+  }
+  props.push_back({"BTerms", bterms});
+
+  SelectionSet nets;
+  for (auto net : block->getNets()) {
+    nets.insert(gui->makeSelected(net));
+  }
+  props.push_back({"Nets", nets});
+
+  SelectionSet regions;
+  for (auto region : block->getRegions()) {
+    regions.insert(gui->makeSelected(region));
+  }
+  props.push_back({"Regions", regions});
+
+  SelectionSet insts;
+  for (auto inst : block->getInsts()) {
+    insts.insert(gui->makeSelected(inst));
+  }
+  props.push_back({"Insts", insts});
+
+  SelectionSet iterms;
+  for (auto iterm : block->getITerms()) {
+    iterms.insert(gui->makeSelected(iterm));
+  }
+  props.push_back({"ITerms", iterms});
+
+  SelectionSet blockages;
+  for (auto blockage : block->getBlockages()) {
+    blockages.insert(gui->makeSelected(blockage));
+  }
+  props.push_back({"Blockages", blockages});
+
+  SelectionSet obstructions;
+  for (auto obstruction : block->getObstructions()) {
+    obstructions.insert(gui->makeSelected(obstruction));
+  }
+  props.push_back({"Obstructions", obstructions});
+
+  SelectionSet rows;
+  for (auto row : block->getRows()) {
+    rows.insert(gui->makeSelected(row));
+  }
+  props.push_back({"Rows", rows});
+
```

Comment:
What about `getTopModule()`, `getMOdules()`, and `getChildren()`, `getVias()`
core and die area?

Reply:
I agree there are a bunch more but I don't want to overload this PR.  core and die area seem pretty easy.

Reply:
Should I add the all or just core and die area?